pkg/stanza/operator/parser/syslog/syslog.go

```diff
@@ -279,3 +319,47 @@ func postprocess(e *entry.Entry) error {
 
 	return nil
 }
+
+func validateNonTransparentFramingTrailer(trailer *string) error {
+	if trailer == nil {
+		return nil
+	}
+
+	if *trailer != NULTrailer && *trailer != LFTrailer {
+		return fmt.Errorf("invalid non_transparent_framing_trailer '%s'. Must be either 'LF' or 'NUL'", *trailer)
+	}
+
+	return nil
```

Comment:
I did the nil check as this function has no guarantee that `trailer` isn't nil from the perspective of this function. And dereferencing a nil pointer results in a panic. It's just a safety check to ensure there's no unintended panic at some point. I can change it if you like.

Reply:
I think it's ok since this function is called for one very specific case. In fact, it'd be ok to remove the func entirely and move this code directly into the case that calls it.